#161966897 Test Login via Social Authentication

codeplus254 opened the pull request:
#### What does this PR do?
Tests the login via Google, Twitter or Facebook
#### Description of Task to be completed?
Write failing tests before the login feature is implemented. 
This will be done by updating the serializers.py, urls.py and views.py
#### How should this be manually tested?

- After cloning the repo, CD into it. 

- Create and start your virtual environment 

- Install dependencies. `pip3 install -r requirements.txt`

- Define the following environment variables and place them in a `.env` file in the root directory

`export DATABASE_USER='postgres'
export DATABASE_PASSWORD='postgres'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='test_db' `

- Add the variables into the virtual environment you just created. `source .env`

- run `python3 manage.py test`. If the tests cannot be found automatically, you'll need to add the url after the word test (  `python3 manage.py test authors.apps.authentication.tests.social_authentication_tests` )
#### Any background context you want to provide?
The user will have options for logging in directly into the app through Facebook, Google or Twitter
#### What are the relevant pivotal tracker stories?
[#161966897](https://www.pivotaltracker.com/story/show/161966897)
#### Screenshots (if appropriate)
<img width="1440" alt="screenshot 2018-12-07 at 09 26 23" src="https://user-images.githubusercontent.com/16904946/49631324-94e95b80-fa02-11e8-8f91-a7542ab48e65.png">
<img width="1440" alt="screenshot 2018-12-07 at 09 27 18" src="https://user-images.githubusercontent.com/16904946/49631331-9e72c380-fa02-11e8-89fd-db77daae77ad.png">

Commit:
chore(test-registration): Test user registration

-Add .DS_Store,.vscode and virtual folder to .gitignore
-Create test for user registration
-Fix routing errors
-Add input validation tests
-Add test to check for duplicate data
-Test for missing data, valid username and password
-Add tests for missing parameters
-Configure code to pep8 standards
-Generate user token

[Starts #161966897]

codeplus254 committed:
chore(login): test user login process after they submit email and password

- ensure the login process (no social authentication) is adequately tested

[Delivers #161966897]

ThaDeveloper commented:
Good thing there are zero errors in the tests. Include how to install dependencies and run migrations.

codeplus254 committed:
chore(social authentication): test social authentication

- ensure the login process via Facebook, Twitter and Google is adequately tested.

[Delivers #161966897]

r-wambui commented:
@codeplus254 Please rebase develop

r-wambui commented on .gitignore:
```diff
@@ -92,3 +94,6 @@ ENV/
 db.sqlite3
+
+#Django Migrations
+0001_initial.py
```
Why did you add migrations to gitignore?

codeplus254 replied:
@r-wambui the PR couldn't be merged automatically with that file in...plus it's autogenerated by django

r-wambui replied:
But we do not ignore migrations. You could have looked for a way to solve the conflicts.

codeplus254 replied:
Ok let me find a way around this. :)

r-wambui commented on authors/apps/authentication/tests/social_authentication_tests.py:
```diff
@@ -0,0 +1,75 @@
+"""This module runs tests for user login process via Facebook, Twitter or Google"""
+import json
```
json is imported but not used

codeplus254 replied:
I don't know what happened. Pylint had noted this and I changed but the changes were lost in some git process I ran

r-wambui commented on authors/apps/authentication/tests/social_authentication_tests.py:
```diff
@@ -0,0 +1,75 @@
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_facebook_login_no_tokens(self):
+        """this method test that a user cannot login via Facebook if there's no access_key and access_key_secret"""
```
Good use of docstrings. However you did not follow pep8. 
https://www.python.org/dev/peps/pep-0008/#maximum-line-length

codeplus254 replied:
Sorry, I had Pylinted the file but somewhere in the rebase or squash or sth I lost the changes.

r-wambui commented on authors/settings.py:
```diff
@@ -43,7 +43,16 @@
     'authors.apps.core',
     'authors.apps.profiles',
+
+    'social_django'
```
Should we be having social-django in the requirements.txt as well?

codeplus254 replied:
Yes I will add it.

rkemmy commented:
@codeplus254 you can add an  `__init__.py` file to your test folder so that it is considered a package and you can then run tests using `python manage.py test`

codeplus254 committed:
Merge pull request #26 from andela/ch-user-login-tests-161966897

Merging log in into user-registration.

codeplus254 committed:
Merge branch 'ch-user-registration-tests-161966897' into ch-social-authentication-tests-161966897
